### ISSUE TITLE
CURL_TIMEOUT_MS not working correctly in Guzzle

### ISSUE THREAD

**Issue by tomjankes** [guzzle/guzzle#527]
Hello,

we have issue with setting read timeout in Guzzle. Curl gets correct read timeout, but timeout is reached only after one second.

We suspect it is due to selectTimeout passed to curl_multi_select being one second (at least changing this value to lower fixed the problem for us - https://github.com/guzzle/guzzle/blob/master/src/Guzzle/Http/Curl/CurlMulti.php#L232)

I created small project to show the behaviour:
https://github.com/tomjankes/guzzle-timeout-reporoduce/

Maybe instead of using hardcoded `$selectTimeout = 1`, we should use something dynamic like `$selectTimeout = min($selectTimeout + 0.1, 1);` or some other (exponential?) function.

Bottom line is, after setting CURL_TIMEOUT_MS to something lower than 1s the expected behaviour would be for Guzzle to throw timout exception before 1s passes.
